### PR TITLE
authz: Swap to using the correct TypedConfig in audit logger parsing

### PR DESCRIPTION
**Pull request by gtcooke94**
We parse custom configs and need to parse them a little more strictly than what was written with anypb, we use the `TypedConfig` that has a `TypeUrl` and `Value`, where the value is the custom config

RELEASE NOTES: none

### CODE REVIEW

File: authz/rbac_translator.go [hunk 1]
```diff
@@ -36,6 +37,8 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+const typedUrlPrefix = "grpc.authz.audit_logging/"
```
Comment (rockspore) [inline authz/rbac_translator.go:40]: Suggest adding a comment that this is used to convert the config into a typedstruct.

File: authz/rbac_translator.go [hunk 2]
```diff
@@ -302,11 +305,20 @@ func (options *auditLoggingOptions) toProtos() (allow *v3rbacpb.RBAC_AuditLoggin
...
+		if err != nil {
+			return nil, nil, fmt.Errorf("error creating custom config: %v", err)
+		}
+		customConfig, err := anypb.New(typedStruct)
 		if err != nil {
 			return nil, nil, fmt.Errorf("error parsing custom audit logger config: %v", err)
 		}
-		logger := &v3corepb.TypedExtensionConfig{Name: config.Name, TypedConfig: customConfig}
+
+		logger := &v3corepb.TypedExtensionConfig{Name: config.Name /*this doesn't matter*/, TypedConfig: customConfig}
```
Comment (rockspore) [inline authz/rbac_translator.go:321]: nit: Maybe elaborate a little more on why this does not matter.

File: authz/rbac_translator.go [hunk 3]
```diff
@@ -302,11 +305,20 @@ func (options *auditLoggingOptions) toProtos() (allow *v3rbacpb.RBAC_AuditLoggin
 		if config.Config == nil {
 			return nil, nil, fmt.Errorf("AuditLogger Config field cannot be nil")
 		}
-		customConfig, err := anypb.New(config.Config)
+
+		typedStruct := &v1typepb.TypedStruct{
+			TypeUrl: typedUrlPrefix + config.Name,
+			Value:   config.Config,
+		}
+		if err != nil {
```
Comment (rockspore) [inline authz/rbac_translator.go:313]: There is no new error to check here.

**Review by rockspore — APPROVED** [review]
LGTM besides the comments. Thanks for working on it!

File: authz/rbac_translator.go (hunk 2 above)
Comment (gtcooke94) [reply authz/rbac_translator.go:321]: I'm just going to remove that comment actually

File: authz/rbac_translator.go (hunk 1 above)
Comment (gtcooke94) [reply authz/rbac_translator.go:42]: Fixed, PTAL

File: authz/rbac_translator.go (hunk 2 above)
Comment (gtcooke94) [reply authz/rbac_translator.go]: If that is good with you please resolve

**Comment by gtcooke94** [comment]
@dfawley @easwars can one of you spare time for a quick review on this?

**Review by easwars — APPROVED** [review]

**Pull request merged**